scripts/core/counterbasedevent.yml

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (C) 2019-2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Level-Zero Extension APIs for Counter-based Events"
+version: "1.8"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Counter-based Events Extension Name"
+version: "1.4"
+name: $X_EVENT_COUNTER_BASED_EXP_NAME
+value: '"$X_experimental_event_counter_based"'
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Counter-based Events Extension Version(s)"
+version: "1.8"
+name: $x_event_counter_based_exp_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Supported event flags for defining counter-based events. Must "
+class: $xEvent
+name: $x_event_counter_based_exp_flags_t
+etors:
+    - name: IMMEDIATE
+      desc: "Counter-based event is used for immediate command lists (default)"
+    - name: NON_REGULAR
```

Comment:
do you mean NON_IMMEDIATE here?

Reply:
thanks @MichalMrozek . Fixed.

Reply:
do we need two flags? say if we just have IMMEDIATE, then the absence of that flag would mean REGULAR cmdlists?

Reply:
thanks @aravindksg . I was thinking on that. However, it is better to be explicit. in any case, as you see, we are saying that default value is zero, which is IMMEDAITE, so the user only needs to pass a flag (NON_REGULAR) if they want for regular. 

```cpp
type: $x_event_counter_based_exp_flags_t
      name: flags
      desc: |
            [in] mode flags.
            must be 0 (default)
```